Sources/TSPL/TSPL.docc/ReferenceManual/Attributes.md

```diff
@@ -1362,11 +1364,39 @@ If your result-building methods
 don't specify a type for `Expression` or `FinalResult`,
 they default to being the same as `Component`.
 
-The result-building methods are as follows:
+The block-building methods are as follows:
 
 - term `static func buildBlock(_ components: Component...) -> Component`:
   Combines an array of partial results into a single partial result.
-  A result builder must implement this method.
+
+- term `static func buildPartialBlock(first: Component) -> Component`:
+  Builds a partial result component from the first component.
+  Implement both this method and `buildPartialBlock(accumulated:next:)`
+  to support building blocks one component at a time.
+  Compared to `buildBlock(_:)`,
+  this approach reduces the need for generic overloads
+  that handle different numbers of arguments.
+
+- term `static func buildPartialBlock(accumulated: Component, next: Component) -> Component`:
+  Builds a partial result component
+  by combining an accumulated component with a new component.
+  Implement both this method and `buildPartialBlock(first:)`
+  to support building blocks one component at a time.
+  Compared to `buildBlock(_:)`,
+  this approach reduces the need for generic overloads
+  that handle different numbers of arguments.
+
+If a result builder implements all three of the block-building methods listed above,
+availability determines which method gets used.
+If the availability of the enclosing declaration is the same as
+the availability of `buildPartialBlock(first:)`
+and `buildPartialBlock(accumulated:next:)`,
+the build-partial-block methods are used.
+If the enclosing declaration is also available in older versions of the library,
+the build-partial-block methods are used.
+Otherwise, `buildBlock(_:)` is used.
```

Comment:
@rxwei The SE proposal describes the availability behavior as "greater than or equal", but I'm not entirely sure what that means.  Is this rephrased description is correct?

Reply:
Yeah that's unclear wording. "Greater than or equal" is talking about the version number (e.g. the `13` in `@available(macOS 13, ...)` being greater. So the enclosing declaration must be not available in an OS older than `buildPartialBlock`'s availability, in order for `buildPartialBlock` to be used.

I think that we'd want the opposite of "If the enclosing declaration .... Otherwise ..." above:

If the enclosing declaration is also available in older versions of the library,
`buildBlock(_:)` is used. Otherwise, build-partial-block methods are used.